dashboard/app/models/blockly.rb

```diff
@@ -37,7 +39,8 @@ class Blockly < Level
 
   # These serialized fields will be serialized/deserialized as straight XML
   def xml_blocks
-    %w(start_blocks toolbox_blocks required_blocks)
+    return %w(start_blocks toolbox_blocks required_blocks) unless self.game.try(:uses_droplet?)
```

Comment:
Yes, in this PR I have a slightly better fix for this where xml_blocks is changed only when overridden by applab.rb: https://github.com/code-dot-org/code-dot-org/pull/1373

But this doesn't take care of the other apps that can be either blockly or droplet. I was hoping that editCode could simply be an attribute on Maze, Studio, Artist, etc. but with this other approach I may have to create separate classes for each depending on blockly vs. droplet.